packages/components/src/FilterDateRangePicker/FilterDateRangePicker.spec.tsx

```diff
@@ -83,5 +98,31 @@ describe("<FilterDateRangePicker />", () => {
         expect(screen.getByText("May 2022")).toBeVisible()
       })
     })
+
+    it("should not show a date range in the button if the selected range is not valid", async () => {
+      render(
+        <FilterDateRangePickerWrapper defaultMonth={new Date("2022-05-01")} />
+      )
+      expect(screen.queryByRole("dialog")).not.toBeInTheDocument()
+
+      const filterButton = screen.getByRole("button", {
+        name: "Dates",
+      })
+
+      await user.click(filterButton)
+
+      const inputEndDate = screen.getByLabelText("Date to")
+
+      await waitFor(() => {
+        expect(inputEndDate).toBeVisible()
+      })
+
+      await user.clear(inputEndDate)
+      await user.type(inputEndDate, "01/04/2022")
+
+      await user.click(document.body)
+
+      expect(filterButton).toBeVisible()
```

Comment:
The button hasn't been removed from the dom, so this will still be valid even if the text has changed. You want to check the text instead.